zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -643,33 +644,37 @@ protected void syncWithLeader(long newLeaderZxid) throws Exception {
                         self.setLastSeenQuorumVerifier(qv, true);
                     }
 
+                    packetsNotLogged.add(pif);
                     packetsNotCommitted.add(pif);
                     break;
                 case Leader.COMMIT:
                 case Leader.COMMITANDACTIVATE:
                     pif = packetsNotCommitted.peekFirst();
-                    if (pif.hdr.getZxid() == qp.getZxid() && qp.getType() == Leader.COMMITANDACTIVATE) {
-                        QuorumVerifier qv = self.configFromString(new String(((SetDataTxn) pif.rec).getData(), UTF_8));
-                        boolean majorChange = self.processReconfig(
-                            qv,
-                            ByteBuffer.wrap(qp.getData()).getLong(), qp.getZxid(),
-                            true);
-                        if (majorChange) {
+                    if (pif.hdr.getZxid() != qp.getZxid()) {
+                        LOG.warn(
+                            "Committing 0x{}, but next proposal is 0x{}",
+                            Long.toHexString(qp.getZxid()),
+                            Long.toHexString(pif.hdr.getZxid()));
+                    } else {
```

Comment:
I'm not sure if this branch is correct, because in the old logic if `writeToTxnLog == true`, then the packet is added to `packetsCommitted` no matter if zxid matches with the next proposal or not.

In your new logic, if I understand it right, if `zxid` differs you'll quit fast and log the above warning message only.

Is that correct?